ion-schema/src/main/kotlin/com/amazon/ionschema/model/UserReservedFields.kt

```diff
@@ -0,0 +1,11 @@
+package com.amazon.ionschema.model
+
+/**
+ * The collection of field names that are reserved by the user as open content fields.
+ * See relevant section in [ISL 2.0 spec](https://amazon-ion.github.io/ion-schema/docs/isl-2-0/spec#open-content).
+ */
+data class UserReservedFields constructor(
+    val type: List<String> = emptyList(),
+    val header: List<String> = emptyList(),
+    val footer: List<String> = emptyList(),
```

Comment:
hrrm. The problem is that Kotlin's List is read-only but not technically immutable. IIRC, like `internal` this is something that can be circumvented with java code. Not sure how protective we want to get, just making sure you're aware of it.

Reply:
I am aware. If we're concerned about this, I could bring in the `kotlinx.collections.immutable` library.